### ISSUE TITLE
在 onSuccess 回调函数里获取其他useState的值，一直都是默认初始化的值

### ISSUE THREAD

**Issue by anhao** [shahradelahi/react-geetest#2]
在 onSuccess 回调函数里获取其他useState的值，一直都是默认初始化的值

```
   const [registerObj, setRegisterObj] = useState({
        username: "",
        password: "",
        email: "",
        code: "",
    })

   const handleSendCode = ()=>{
        console.log(captchaRef)
        if (!captchaRef.current){
            toast.error("滑块验证码加载失败，请刷新页面，或者检测是否被浏览器广告拦截插件拦截")
            return
        }
        console.log(registerObj);//这里获取的值一直是默认的初始化的值，是有更改值得，就是放在 geetest 的OnSuccess会一直是默认的值，其他地方调用是正常的
       
        console.log(captchaRef.current.getValidate())
    }

```

**Comment by shahradelahi** [comment]
Hello @anhao,

I don't understand where the problem lies. The `onSuccess` function is not initialized by default, so please provide more context.

When you use the `GeeTest` component, it retrieves the client-side validation result. Instead of using this:

```jsx
console.log(captchaRef.current.getValidate())
```

You can use this:

```tsx
<GeeTest
 // ...
 onSuccess={(result) => console.log('success. result: ', result)}
/>
```

Please let me know if you need any further assistance.

**Comment by anhao** [comment]
```
  const INITIAL_REGISTER_OBJ = {
        username: "",
        password: "",
        email: "",
        code: "",
    }
    const [registerObj, setRegisterObj] = useSetState(INITIAL_REGISTER_OBJ)

    const handleSendCode = (result)=>{
        console.log(registerObj);
       //  The default value has always been obtained here, but it has actually changed.
    }



<GeeTest
captchaId={captchaConfig.captchaId}
product={'bind'}
onSuccess={handleSendCode}
 >
 <button type='button' className={'btn btn-primary ' + (!canSend ? 'disabled':'')} disabled={!canSend} >{canSend ? '获取验证码' : '重新获取 '+countdown + 's'}</button>
 </GeeTest>



```

**Comment by shahradelahi** [comment]
The problem has been resolved in `v1.1.1`. To fix it, kindly update to the latest version by running the following command:
 
```bash
npm i react-geetest-v4@latest
```
